### PR TITLE
Pull Members Up: A couple of fixes to pull to base class as abstract

### PR DESCRIPTION
**Pull request by YairHalberstadt**
See #34235

Make methods overrides when pulling to base class with the make abstract option.

Remove abstract methods when pulling to base class with the make abstract option.

### CODE REVIEW

File: src/Features/Core/Portable/PullMemberUp/MembersPuller.cs [hunk 1]
```diff
@@ -254,20 +254,28 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
...
                     {
-                        var originalMemberEditor = await solutionEditor.GetDocumentEditorAsync(
-                            solution.GetDocumentId(syntax.SyntaxTree),
-                            cancellationToken).ConfigureAwait(false);
                         originalMemberEditor.RemoveNode(originalMemberEditor.Generator.GetDeclaration(syntax));
                     }
+                    else
+                    {
+                        var declarationSyntax = originalMemberEditor.Generator.GetDeclaration(syntax);
+                        var generator = SyntaxGenerator.GetGenerator(solution.GetDocument(syntax.SyntaxTree));
```
[the remark this answers is not in the document]
Comment (YairHalberstadt) [reply src/Features/Core/Portable/PullMemberUp/MembersPuller.cs]: I didn't know that method existed :-)
I've made that change now.

File: src/Features/Core/Portable/PullMemberUp/MembersPuller.cs [hunk 2]
```diff
@@ -254,20 +254,26 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
...
                     {
-                        var originalMemberEditor = await solutionEditor.GetDocumentEditorAsync(
-                            solution.GetDocumentId(syntax.SyntaxTree),
-                            cancellationToken).ConfigureAwait(false);
                         originalMemberEditor.RemoveNode(originalMemberEditor.Generator.GetDeclaration(syntax));
                     }
+                    else
+                    {
+                        var declarationSyntax = originalMemberEditor.Generator.GetDeclaration(syntax);
+                        originalMemberEditor.ReplaceNode<object>(declarationSyntax, (node, generator, _) => generator.WithModifiers(node, DeclarationModifiers.Override), null);
```
Comment (Cosifne) [inline src/Features/Core/Portable/PullMemberUp/MembersPuller.cs]: Why not just use the normal overload version of ReplaceNode here ?(i.e. the one without the generics)

Comment (YairHalberstadt) [reply src/Features/Core/Portable/PullMemberUp/MembersPuller.cs]: Oops. I think I just got confused by intellisense :-)

**Comment by YairHalberstadt** [comment]
I was just wondering, what is the process of getting this merged now that it is approved?

Thanks.

**Comment by Cosifne** [comment]
@jinujoseph and @vatsalyaagrawal 
Is this OK to merge to master now?

**Comment by jinujoseph** [comment]
@Cosifne feel free to merge once it's all green

**Comment by Cosifne** [comment]
@YairHalberstadt  Thank you. I'll merge this PR

**Comment by Cosifne** [comment]
> 
> 
> I was just wondering, what is the process of getting this merged now that it is approved?
> 
> Thanks.

I'll merge this after CI completes

**Pull request merged**